kotlin/display-feature-layers/src/main/java/com/esri/arcgisruntime/sample/displayfeaturelayers/MainActivity.kt

```diff
@@ -0,0 +1,263 @@
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgisruntime.sample.displayfeaturelayers
+
+import android.os.Bundle
+import android.util.Log
+import android.widget.ArrayAdapter
+import android.widget.AutoCompleteTextView
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import com.esri.arcgisruntime.ArcGISRuntimeEnvironment
+import com.esri.arcgisruntime.data.GeoPackage
+import com.esri.arcgisruntime.data.Geodatabase
+import com.esri.arcgisruntime.data.ServiceFeatureTable
+import com.esri.arcgisruntime.data.ShapefileFeatureTable
+import com.esri.arcgisruntime.layers.FeatureLayer
+import com.esri.arcgisruntime.loadable.LoadStatus
+import com.esri.arcgisruntime.mapping.ArcGISMap
+import com.esri.arcgisruntime.mapping.BasemapStyle
+import com.esri.arcgisruntime.mapping.Viewpoint
+import com.esri.arcgisruntime.mapping.view.MapView
+import com.esri.arcgisruntime.portal.Portal
+import com.esri.arcgisruntime.portal.PortalItem
+import com.esri.arcgisruntime.sample.displayfeaturelayers.databinding.ActivityMainBinding
+import com.esri.arcgisruntime.security.UserCredential
+import java.io.File
+
+
+class MainActivity : AppCompatActivity() {
+
+    // enum to keep track of the selected source to display the feature layer
+    enum class FeatureLayerSource(val menuPosition: Int) {
+        SERVICE_FEATURE_TABLE(0),
+        PORTAL_ITEM(1),
+        GEODATABASE(2),
+        GEOPACKAGE(3),
+        SHAPEFILE(4)
+    }
+
+    private val TAG = MainActivity::class.java.simpleName
+
+    // keeps track of the previously selected feature layer source
+    private var previousSource: FeatureLayerSource? = null
+
+    private val activityMainBinding by lazy {
+        ActivityMainBinding.inflate(layoutInflater)
+    }
+
+    private val mapView: MapView by lazy {
+        activityMainBinding.mapView
+    }
+
+    private val bottomListItems: AutoCompleteTextView by lazy {
+        activityMainBinding.bottomListItems
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(activityMainBinding.root)
+
+        // authentication with an API key or named user is required to
+        // access basemaps and other location services
+        ArcGISRuntimeEnvironment.setApiKey(BuildConfig.API_KEY)
+
+        // create a map with the BasemapStyle topographic
+        val map = ArcGISMap(BasemapStyle.ARCGIS_TOPOGRAPHIC)
+        // set the map to be displayed in the layout's MapView
+        mapView.map = map
+        setUpBottomUI()
+    }
+
+    private fun setUpBottomUI() {
```

Comment:
Since this is mostly UI stuff, can we move it to the bottom so the Runtime code showing FeatureLayer creation is front and center?